drivers/timer/riscv_machine_timer.c

```diff
@@ -21,12 +21,17 @@
 static struct k_spinlock lock;
 static uint64_t last_count;
 
+static uint64_t get_hart_mtimecmp(void)
+{
+	return RISCV_MTIMECMP_BASE + (_current_cpu->id * 8);
```

Comment:
I didn't know there was one of these per core.  Are we guaranteed that all CPU clocks are synchronized?

Reply:
There's a single `mtime` per system to count the time, but `mtimecmp` is used to trigger a timer interrupt.